player/loadfile.c

```diff
@@ -1467,8 +1468,10 @@ static int reinit_complex_filters(struct MPContext *mpctx, bool force_uninit)
     cleanup_deassociated_complex_filters(mpctx);
 
     if (mpctx->playback_initialized) {
-        for (int n = 0; n < mpctx->num_tracks; n++)
+        for (int n = 0; n < mpctx->num_tracks; n++) {
+            mpctx->tracks[n]->stream->max_packets = max_packets;
```

Comment:
`mpctx->tracks[n]->stream->packets_read = 0;` here fixes enabling and disabling `--lavfi-complex` multiple times.

Reply:
Actually I can simplify this to just a single boolean.